src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -4,19 +4,21 @@ import * as React from 'react';
 
 import BaseChannelItem from './BaseChannelItem';
 import useChatUtilsHook from '../../hooks/core/chat/useChatUtilsHook';
-import useProfileHook from '../../hooks/core/profile/useProfileHook';
+import useFollowUserV2Hook from '../../hooks/core/users/useFollowUserV2Hook';
+import useUserAuthHook from '../../hooks/core/auth/useUserAuthHook';
 import {BaseChannelItemTypeProps} from '../../../types/component/AnonymousChat/BaseChannelItem.types';
 import {ChannelList} from '../../../types/database/schema/ChannelList.types';
 import {MessageChannelItemProps} from '../../../types/component/AnonymousChat/MessageChannelItem.types';
 import {calculateTime} from '../../utils/time';
 
 const MessageChannelItem: (props: MessageChannelItemProps) => React.ReactElement = ({
   item,
-  onChannelPressed,
-  hasFollowButton = false,
-  handleFollow
+  onChannelPressed
 }) => {
-  const {signedProfileId} = useProfileHook();
+  const {signedProfileId} = useUserAuthHook();
+  const {followStatus, getOtherProfileFollowStatus, followUserAction} =
+    useFollowUserV2Hook('signed');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new hooks `useFollowUserV2Hook` and `useUserAuthHook` are imported and used correctly. However, it's important to ensure that these hooks are properly tested and handle all edge cases since they are replacing the old `useProfileHook`. Also, make sure that the 'signed' argument passed to `useFollowUserV2Hook` is valid and appropriate for this context.

```diff
-  const {signedProfileId} = useProfileHook();
+  const {signedProfileId} = useUserAuthHook();
+  const {followStatus, getOtherProfileFollowStatus, followUserAction} =
+    useFollowUserV2Hook('signed');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->